modules/compiler/targets/host/source/kernel.cpp

```diff
@@ -388,8 +388,62 @@ HostKernel::lookupOrCreateOptimizedKernel(std::array<size_t, 3> local_size) {
       // Compiling the kernel may touch the global LLVM state
       std::lock_guard<std::mutex> globalLock(
           compiler::utils::getLLVMGlobalMutex());
-      auto sym = target.orc_engine->lookup(*jd, unique_name);
-      if (auto err = sym.takeError()) {
+
+      // We cannot safely look up any symbol inside a CrashRecoveryContext
+      // because the CRC handles errors by a longjmp back to safety, skipping
+      // over destructors of objects that do need to be destroyed. We do so
+      // anyway because the effect is less bad than crashing right away.
+      std::promise<uint64_t> promise;
+      llvm::Error err = llvm::Error::success();
+      llvm::cantFail(std::move(err));
+
+      auto &ES = target.orc_engine->getExecutionSession();
+      auto SO = makeJITDylibSearchOrder(
+          &*jd, llvm::orc::JITDylibLookupFlags::MatchAllSymbols);
+      auto Name = ES.intern(target.orc_engine->mangle(unique_name));
```

Comment:
I know we're not in a consistent place across the project, but this function seems to have settled on a naming scheme which `Name` isn't adhering to (in my opinion).

Obviously it's arbitrary but I think `CRC` is probably fine for its conciseness, but I think `name` and `names` and `notify_complete` might suit better? I realise we have `globalLock` above too. I'm not sure about `ES` or `SO`. Ugh..

Reply:
Ah, right, this is basically copied and modified from LLVM code and I didn't adjust the LLVM naming of things. Will do.

Reply:
Changed everything to start lowercase to mostly fit the existing style. Given that this is a function named `lookupOrCreateOptimizedKernel`, I think the style for multi-word names needs to be camel case even if not everything else in the function follows that same style, does that sound okay to you?